FormalConjectures/ErdosProblems/44.lean

```diff
@@ -0,0 +1,114 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE/2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+import FormalConjectures.ForMathlib.Combinatorics.Basic
+
+/-!
+# Erdős Problem 44: Extending Sidon Sets
+
+*Reference:* [erdosproblems.com/44](https://www.erdosproblems.com/44)
+
+Let N ≥ 1 and `A ⊆ {1,…,N}` be a Sidon set. Is it true that, for any ε > 0,
+there exist M = M(ε) and `B ⊆ {N+1,…,M}` such that `A ∪ B ⊆ {1,…,M}` is a Sidon set
+of size at least `(1−ε)M^{1/2}`?
+
+This problem asks whether any Sidon set can be extended to achieve a density
+arbitrarily close to the optimal density for Sidon sets.
+-/
+
+open Function Set
+
+/-- The maximum size of a Sidon set in `{1, ..., N}`. -/
+noncomputable def maxSidonSetSize (N : ℕ) : ℕ :=
+  sSup {A.card | (A : Finset ℕ) (_ : A ⊆ Finset.Icc 1 N) (_ : IsSidon A.toSet)}
+
+/-- The optimal density for Sidon sets is approximately `√N`. -/
+@[category API, AMS 5 11]
+theorem maxSidonSetSize_bound (N : ℕ) (hN : 1 ≤ N) :
+    maxSidonSetSize N ≤ N.sqrt + 1 := by
+  sorry
+
+/--
+A Sidon set can be extended to achieve density close to optimal.
```

Comment:
I would reword this closer to how it is written on the website. In particular it should be phrased as a question rather than an assertion (especially since you are using the `answer(sorry)` formulation.